pygmt/helpers/tempfile.py

```diff
@@ -126,18 +126,27 @@ def tempfile_from_geojson(geojson):
         E.g. '1a2b3c4d5e6.gmt'.
     """
     with GMTTempFile(suffix=".gmt") as tmpfile:
+        # pylint: disable=import-outside-toplevel
+        import geopandas as gpd
+
         os.remove(tmpfile.name)  # ensure file is deleted first
         ogrgmt_kwargs = {"filename": tmpfile.name, "driver": "OGR_GMT", "mode": "w"}
         try:
+            # Workaround to map int/int64 to int32
+            # https://github.com/geopandas/geopandas/issues/967#issuecomment-842877704
+            # https://github.com/GenericMappingTools/pygmt/issues/2497
+            schema = gpd.io.file.infer_schema(geojson)
```

Comment:
Hmm, looking into this [`infer_schema`](https://github.com/geopandas/geopandas/blob/35463adda663dfa57c9809865ebbb15448823278/geopandas/io/file.py#L625) function and https://github.com/geopandas/geopandas/issues/967#issuecomment-842999302 a bit more, I almost think part of the fix should go upstream into modifying `infer_schema` to output int32 values when the column is of int32 dtype, instead of forcing things to int/int64.

Of course, we'll still need to force the OGR_GMT output to int32 in PyGMT, but the code would be much simplified with some upstream fixes.

Reply:
> Hmm, looking into this [`infer_schema`](https://github.com/geopandas/geopandas/blob/35463adda663dfa57c9809865ebbb15448823278/geopandas/io/file.py#L625) function and [geopandas/geopandas#967 (comment)](https://github.com/geopandas/geopandas/issues/967#issuecomment-842999302) a bit more, I almost think part of the fix should go upstream into modifying `infer_schema` to output int32 values when the column is of int32 dtype, instead of forcing things to int/int64.

There is already a open PR https://github.com/geopandas/geopandas/pull/2265, but it has been inactive for one year.

Reply:
> There is already an open PR [geopandas/geopandas#2265](https://github.com/geopandas/geopandas/pull/2265), but it has been inactive for one year.

Ok, and I saw https://github.com/geopandas/geopandas/pull/2464 as well, also inactive :laughing:

Started a small PR at https://github.com/geopandas/geopandas/pull/2950 to just handle int32, let's see if their test suite passes, I took a look at https://fiona.readthedocs.io/en/latest/manual.html#keeping-schemas-simple and `fiona.FIELD_TYPES_MAP` seem to have the following mapping:

```
{'int32': int,
 'float': float,
 'str': str,
 'date': fiona.rfc3339.FionaDateType,
 'time': fiona.rfc3339.FionaTimeType,
 'datetime': fiona.rfc3339.FionaDateTimeType,
 'bytes': bytes,
 'int64': int,
 'int': int,
 'List[str]': typing.List[str]}
```
So not sure if the geopandas default driver (ESRI Shapefile?) supports int32 schema. It it doesn't and the test suite breaks, then we might need to handle everything fully in PyGMT.

Reply:
It's still unclear if there will be any changes in GDAL, GeoPandas, Fiona or GMT, but as we are still using old versions of these packages, I believe we still need these workarounds in the next few years, right?

Reply:
So the patch in geopandas has been merged at https://github.com/geopandas/geopandas/pull/2950, but will need to wait for geopandas v0.13.3 or newer to be released. Even so, int64 will still need to be converted to int32, so yes, we'll need to keep this workaround for a while.